rust/src/lib/ovn.rs

```diff
@@ -0,0 +1,268 @@
+                mappings_ext_id_value: None,
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+/// Global OVN bridge mapping configuration. Example yaml output of [crate::NetworkState]:
+/// ```yml
+/// ---
+/// ovn:
+///   bridge-mappings:
+///   - localnet: tenantblue
+///     bridge: ovsbr1
+///     state: present
+///   - localnet: tenantred
+///     bridge: ovsbr1
+///     state: absent
+/// ```
+pub struct OvnBridgeMapping {
+    pub localnet: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub state: Option<OvnBridgeMappingState>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bridge: Option<String>,
+}
+
+fn value_to_ovn_bridge_mapping(
+    value: &serde_json::Value,
+) -> Result<Option<Vec<OvnBridgeMapping>>, serde_json::Error> {
+    let mut ret: Vec<OvnBridgeMapping> = Vec::new();
+    if let Some(value) = value.as_array() {
+        for v in value.iter() {
+            match OvnBridgeMapping::deserialize(v) {
+                Ok(mapping) => ret.push(mapping),
+                Err(err) => return Err(err),
+            }
+        }
+    }
+    if ret.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(ret))
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct OvnBridgeMappingError {
+    mapping_wannabe: String,
+}
+
+impl std::error::Error for OvnBridgeMappingError {}
+impl OvnBridgeMappingError {
+    fn new(reason: &str) -> Self {
+        Self {
+            mapping_wannabe: reason.to_string(),
+        }
+    }
+}
+impl fmt::Display for OvnBridgeMappingError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "expected `<localnet>:<bridge>`, got: {}",
+            self.mapping_wannabe
+        )
+    }
+}
+
+impl FromStr for OvnBridgeMapping {
+    type Err = OvnBridgeMappingError;
+    fn from_str(s: &str) -> Result<OvnBridgeMapping, OvnBridgeMappingError> {
+        let vec: Vec<&str> = s.split(':').collect();
+        if vec.len() != 2 {
+            return Err(OvnBridgeMappingError::new(s));
+        }
+        let physnet: String = vec[0].to_string();
+        let bridge: String = vec[1].to_string();
+        if physnet.is_empty() || bridge.is_empty() {
+            return Err(OvnBridgeMappingError::new(s));
+        }
+        Ok(OvnBridgeMapping {
+            localnet: physnet,
+            bridge: Some(bridge),
+            state: Some(OvnBridgeMappingState::Present),
+        })
+    }
+}
+
+impl ToString for OvnBridgeMapping {
+    fn to_string(&self) -> String {
+        format!(
+            "{}:{}",
+            self.localnet,
+            self.bridge.clone().unwrap_or("".to_string())
+        )
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase", deny_unknown_fields)]
+#[non_exhaustive]
+pub enum OvnBridgeMappingState {
+    Present,
+    Absent,
```

Comment:
In my test, this YAML is sufficient enough to delete a map:

```yml
ovn:
  bridge-mappings:
    - localnet: red
      state: absent
```

If is on purpose, please add public document for this usage.

Reply:
It is (enough to delete, and on purpose).

I'll add the public doc.

Reply:
Done.